packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -142,6 +144,12 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkCrossReferenceSyntax(crossRef: ast.CrossReference, accept: ValidationAcceptor): void {
+        if (crossRef.terminal && crossRef.$cstNode?.text.includes('|')) {
```

Comment:
You shouldn't use `$cstNode.text` because that includes hidden tokens such as comments. The following cross-ref definition also triggers the validation error although it's syntactically valid:
```
[Foo:/*|*/ID]
```
The correct way to solve this would be to analyze the sub-tree of the CST and look for leaf tokens with the text `|`. You can look into `cst-util.ts` whether there is already a useful utility function for that, or add one if it helps.

An alternative would be to include information about the chosen syntax in the AST by changing the grammar grammar:
```
CrossReference returns AbstractElement:
	{CrossReference} '[' type=[ParserRule:ID] ((deprecatedSyntax?='|' | ':') ^terminal=CrossReferenceableTerminal )? ']';
```
Then you can read the `deprecatedSyntax` property to check which syntax was used. This is the simpler solution, and I think it's ok because we don't regard the grammar language AST format as public API.